soundcard/pulseaudio.py

```diff
@@ -332,6 +332,9 @@ class _Recorder(_Stream):
     after it is closed.
 
     """
+    captured_frames = 0
+    captured_data = []
+    last_record = time.time()
```

Comment:
These should be put in the constructor.

Reply:
Also, they should start with an underscore.

Reply:
Since `captured_data` may only ever contain a single block, it should be sufficient to store something like `self._pending_fragment`, and get rid of `captured_frames`, `captured_data`, and `last_record` (More on `last_record` later).